server/src/main/java/org/opensearch/search/labels/RequestLabelingService.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.labels;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.search.labels.rules.Rule;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Service to attach labels to a search request based on pre-defined rules
+ * It evaluate all available rules and generate labels into the thread context.
+ */
+public class RequestLabelingService {
```

Comment:
Probably don't need standalone Service class for this. The logic can be in the SearchRequestLabelingListener.

Reply:
As we discussed, I'm still keeping this service as the centralized place to read and compute labels.

Reply:
This need not be service IMO, but more like helper class. For now, we can move the static method within `SearchRequestLabelingListener`, again not the best way but better than having this "Helper" service